kafka/producer.go

```diff
@@ -405,3 +435,21 @@ func (p *Producer) GetMetadata(topic *string, allTopics bool, timeoutMs int) (*M
 func (p *Producer) QueryWatermarkOffsets(topic string, partition int32, timeoutMs int) (low, high int64, err error) {
 	return queryWatermarkOffsets(p, topic, partition, timeoutMs)
 }
+
+// OffsetsForTimes looks up offsets by timestamp for the given partitions.
+//
+// The returned offset for each partition is the earliest offset whose
+// timestamp is greater than or equal to the given timestamp in the
+// corresponding partition.
+//
+// The timestamps to query are represented as `.Offset` in the `times`
+// argument and the looked up offsets are represented as `.Offset` in the returned
+// `offsets` list.
+//
+// The function will block for at most timeoutMs milliseconds.
+//
+// Duplicate Topic+Partitions are not supported.
+// Per-partition errors may be returned in the `.Error` field.
+func (p *Producer) OffsetsForTimes(times []TopicPartition, timeoutMs int) (offsets []TopicPartition, err error) {
```

Comment:
Looks good.... but since we're in a language that thinks is-a relationships are so bad it doesn't even support them :-), I'll give my usual speil:

`OffsetsForTimes` is not really a method I like having on something called a 'Producer', which I think should be a class very focussed stuff you want to do when producing. In an ideal world, I think it'd be great if librdkafka allowed for creating a client that could be used after construction for both consumption and production (though specific functionality for doing these tasks would be separate), this concept could then be exposed separately in clients, producer / consumer / admin classes could contain (has-a relationship) a client, and a specific client instance could be re-used.

FWIW, in dotnet we just put this on the consumer... but given the inheritance based object modeling, it does make more sense to have this on the producer as well, so this note is not a call to change anything.